src/NHibernate/Persister/Entity/AbstractEntityPersister.cs

```diff
@@ -1673,14 +1673,10 @@ private string PropertySelectFragment(string name, string suffix, ICollection<st
 					}
 					
 					var columnNames = SubclassPropertyColumnNameClosure[index];
-					// Formulas will have all null values
-					if (columnNames.All(o => o == null))
-					{
-						columnNames = SubclassPropertyFormulaTemplateClosure[index];
-					}
 
-					foreach (var columnName in columnNames)
+					for (var i = 0; i < columnNames.Length; i++)
 					{
+						var columnName = columnNames[i] ?? SubclassPropertyFormulaTemplateClosure[index][i];
```

Comment:
Please extract `SubclassPropertyFormulaTemplateClosure[index]` into a field.

Reply:
Ok. But I wonder if it's really improvement. Array index access is super fast (memory offset) and your change involves stack allocation for temp variable.

Reply:
Compiler should be smart enough to optimize this stack allocation. It is more of a readability issue.